website/content/en/docs/kubebuilder/quickstart.md

```diff
@@ -0,0 +1,763 @@
+
+```YAML
+# Adds namespace to all resources.
+namespace: default
+...
+```
+
+Run the following to deploy the operator. This will also install the RBAC manifests from `config/rbac`.
+
+```sh
+$ make deploy IMG=quay.io/example/memcached-operator:v0.0.1
+```
+
+Verify that the memcached-operator is up and running:
+
+```sh
+$ kubectl get deployment
+NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
+memcached-operator-controller-manager   1/1     1            1           8m
+```
+
+### 3. Deploy your Operator with the Operator Lifecycle Manager (OLM)
+
+OLM will manage creation of most if not all resources required to run your operator, using a bit of setup from other `operator-sdk` commands. Check out the [docs][cli-run-olm] for more information.
+
+## Create a Memcached CR
+
+Update the sample Memcached CR manifest at `config/samples/cache_v1alpha1_memcached.yaml` and define the `spec` as the following:
+
+```YAML
+apiVersion: cache.example.com/v1alpha1
+kind: Memcached
+metadata:
+  name: memcached-sample
+spec:
+  size: 3
+```
+
+Create the CR:
+
+```sh
+$ kubectl apply -f config/samples/cache_v1alpha1_memcached.yaml
+```
+
+Ensure that the memcached operator creates the deployment for the sample CR with the correct size:
+
+```sh
+$ kubectl get deployment
+NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
+memcached-operator-controller-manager   1/1     1            1           8m
+memcached-sample                        3/3     3            3           1m
+```
+
+Check the pods and CR status to confirm the status is updated with the memcached pod names:
+
+```sh
+$ kubectl get pods
+NAME                                  READY     STATUS    RESTARTS   AGE
+memcached-sample-6fd7c98d8-7dqdr      1/1       Running   0          1m
+memcached-sample-6fd7c98d8-g5k7v      1/1       Running   0          1m
+memcached-sample-6fd7c98d8-m7vn7      1/1       Running   0          1m
+```
+
+```sh
+$ kubectl get memcached/memcached-sample -o yaml
+apiVersion: cache.example.com/v1alpha1
+kind: Memcached
+metadata:
+  clusterName: ""
+  creationTimestamp: 2018-03-31T22:51:08Z
+  generation: 0
+  name: memcached-sample
+  namespace: default
+  resourceVersion: "245453"
+  selfLink: /apis/cache.example.com/v1alpha1/namespaces/default/memcacheds/memcached-sample
+  uid: 0026cc97-3536-11e8-bd83-0800274106a1
+spec:
+  size: 3
+status:
+  nodes:
+  - memcached-sample-6fd7c98d8-7dqdr
+  - memcached-sample-6fd7c98d8-g5k7v
+  - memcached-sample-6fd7c98d8-m7vn7
+```
+
+### Update the size
+
+Update `config/samples/cache_v1alpha1_memcached.yaml` to change the `spec.size` field in the Memcached CR from 3 to 4:
+
+```YAML
+apiVersion: cache.example.com/v1alpha1
+kind: Memcached
+metadata:
+  name: memcached-sample
+spec:
+  size: 3
+```
+
+Apply the change:
+
+```
+$ kubectl apply -f config/samples/cache_v1alpha1_memcached.yaml
+```
```

Comment:
Simplify this to use `kubectl patch`?

```
kubectl patch memcached memcached-example -p '{"spec":{"size": 4}}'
```

Reply:
Not positive that this works actually. Might need this instead:
```
kubectl patch memcached memcached-sample -p '[{"op":"replace","path":"/spec/size","value":4}]' --type=json

Reply:
This works too:
```
$ kubectl patch memcached memcached-sample -p '{"spec":{"size": 4}}' --type="merge"
```
Since we're just replacing the field we can specify `--type=merge` which is a [JSON merge patch](https://kubernetes.io/docs/tasks/run-application/update-api-object-kubectl-patch/#use-a-json-merge-patch-to-update-a-deployment).